client.go

```diff
@@ -656,18 +677,21 @@ func (c *Client[TTx]) signalStopComplete(ctx context.Context) {
 		producer.Stop()
 	}
 
-	// Stop all mainline services where stop order isn't important. Contains the
-	// elector and notifier, amongst others.
-	startstop.StopAllParallel(c.services)
-
-	// Once the producers have all finished, we know that completers have at least
-	// enqueued any remaining work. Wait for the completer to finish.
-	//
-	// TODO: there's a risk here that the completer is stuck on a job that won't
-	// complete. We probably need a timeout or way to move on in those cases.
-	c.completer.Wait()
+	// Stop all mainline services where stop order isn't important.
+	startstop.StopAllParallel(append(
+		// This list of services contains the completer, which should always
+		// stop after the producers so that any remaining work that was enqueued
+		// will have a chance to have its state completed as it finishes.
+		//
+		// TODO: there's a risk here that the completer is stuck on a job that
+		// won't complete. We probably need a timeout or way to move on in those
+		// cases.
+		c.services,
 
-	c.queueMaintainer.Stop()
+		// Will only be started if this client was leader, but can tolerate a stop
+		// without having been started.
+		c.queueMaintainer,
```

Comment:
oh yeah, there's probably no reason for this not to stop earlier. Well, maybe it should wait until after the elector has given up any leadership it may hold? 🤔 I guess it's fine if not, worst case they skip a cycle that's about to happen but somebody else will pick it up, right? Is that true of all maintenance services?

Reply:
Yeah, my rationale is basically: the maintenance services are going to be spending most of their time in wait loops anyway, and we expect all maintenance services to be spending most of their time in waits between loops under typical operation. Shutdown should take < 1s once it hits this point, so a new elector and maintenance services should come up quite quickly.

The only service that comes to mind is the periodic job enqueuer because it's essentially stateless now. If it was 59 minutes into about to working a once an hour job, that wait will have to start again on a client elsewhere. But that said, the difference in stopping the queue maintainer a second later will be negligible.

The queue maintainer is a constellation of ~5 services so I did find that having it start to stop in parallel along with these other services did help somewhat in making the top level client tests a little faster in aggregate (since the client is stopped on ~every test). I think overall the tiny tradeoff (if there is one even), is worth it.